active/radiobuttons/RadioButtons_Spec.md

```diff
@@ -169,84 +169,120 @@ If you find that you need wrapping behavior, perhaps a RadioButtons group isn't
 
 # API Details
 
-## RadioButtons
+## RadioButtons IDL
 
-```
-namespace Microsoft.UI.Xaml.Controls
+``` c++
+namespace MU_XC_NAMESPACE
 {
+
+[WUXC_VERSION_PREVIEW]
 [webhosthidden]
 [contentproperty("Items")]
+[MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+[MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]
 unsealed runtimeclass RadioButtons : Windows.UI.Xaml.Controls.Control
 {
     RadioButtons();
 
     Object ItemsSource;
     Windows.Foundation.Collections.IVector<Object> Items{ get; };
-    Windows.UI.Xaml.DataTemplate ItemTemplate;
+    [MUX_DEFAULT_VALUE("winrt::RadioButtonsElementFactory{}")]
+    Object ItemTemplate;
 
-    Windows.UI.Xaml.DependencyObject ContainerFromItem(Object item);
-    Windows.UI.Xaml.DependencyObject ContainerFromIndex(Int32 index);
+    Windows.UI.Xaml.UIElement ContainerFromIndex(Int32 index);
 
+    [MUX_DEFAULT_VALUE("-1")]
     Int32 SelectedIndex;
-    Object SelectedItem;
+    Object SelectedItem{ get; };
```

Comment:
Why is this get-only?

Reply:
ItemsRepeater does not  have a way to get a container from a data object. This is by design as a data object might be duplicated in your data set and then repeater would not know which items to return.   Technically we can do this by either walking the tree looking for a radiobutton who's content matches the supplied item, or by accessing private repeater headers, But in general I think collection controls should not expose selection by data item. given that repeater does not support  it.

Reply:
I think that's a feature gap for Repeater that we should fill. It's ambiguous for ItemsControl too but it's still offered. Binding to SelectedItem is a very popular ask, I don't think we should take that feature away.

Reply:
@jevansaks @StephenLPeters Absolutely! For proper MVVM use cases, being able to bind to the SelectedItem property I feel would be very much appreciated by us customers. At least I would definitely use it much rather than using the SelectedIndex property. And as you have pointed out, history (Treeview SelectedItem property) shows us that there is a good chance an issue will be opened if the RadioButtons control won't ship with a settable (and bindable) SelectedItem property 😉